### PR TITLE
[4.0] Adapt doc blocks of some service providers

### PR DESCRIPTION
**Pull request by laoneo**
Some copy paste errors in service providers.

### CODE REVIEW

File: libraries/src/Service/Provider/Document.php [hunk 1]
```diff
@@ -16,7 +16,7 @@
 use Joomla\DI\ServiceProviderInterface;
 
 /**
- * Service provider for the application's database dependency
+ * Service provider for the document dependency
```
Comment (brianteeman) [inline libraries/src/Service/Provider/Document.php]: Please put "application's" back

Comment (laoneo) [reply libraries/src/Service/Provider/Document.php]: Why? It is not for the application only.

Comment (wilsonge) [reply libraries/src/Service/Provider/Document.php]: So there's two ways of looking at this - one is that the application owns the document (you access it via `JFactory::getApplication()->getDocument()`, the second is that the document object for a given format is the same across applications, whereas for pathway you have specific instances per application (`SiteApplication` and I guess at some point soon for breadcrumbs in the admin template `AdministratorApplication`).

Comment (laoneo) [reply libraries/src/Service/Provider/Document.php]: Agree, it is a dependency for the application but not only, so the text block should not indicate that it is app specific.

**Pull request merged**